evap/staff/templates/staff_questionnaire_index_list.html

```diff
@@ -49,3 +57,24 @@
         </div>
     </div>
 {% endif %}
+
+{% block additional_javascript %}
+    <script type="text/javascript">
+        function changeVisibility(id, visibility) {
+            $.ajax({
+                type: "POST",
+                url: "{% url 'staff:questionnaire_visibility' %}",
+                data: {"questionnaire_id": id, "visibility": visibility},
+                error: function(){ window.alert("{% trans 'The server is not responding.' %}"); },
+                success: function(){ updateState(id, visibility); }
+            });
+        }
+
+        function updateState(id, visibility) {
+            $("#btn-visibility-"+id+"-0").removeClass("active");
```

Comment:
- we're using `onclick` everywhere, so that stays. i prefer that to a listener anyway, because you see at the button what happens when clicking on the button instead of somewhere else, which might be overlooked.
- changing the ids to a data attribute is of course possible - i will make up my mind about that. in general i don't mind addressing each element on the page with a unique id.

Reply:
apart from the onclick listener i applied your suggestions.